src/syscheckd/whodata/win_whodata.c

```diff
@@ -918,6 +991,113 @@ int whodata_audit_start() {
+                        goto error;
+                    }
+                    snprintf(guid_string, 40, "{%08lX-%04hX-%04hX-%02hhX%02hhX-%02hhX%02hhX%02hhX%02hhX%02hhX%02hhX}",
+                                                auditCategoryId.Data1, auditCategoryId.Data2, auditCategoryId.Data3,
+                                                auditCategoryId.Data4[0], auditCategoryId.Data4[1], auditCategoryId.Data4[2], auditCategoryId.Data4[3],
+                                                auditCategoryId.Data4[4], auditCategoryId.Data4[5], auditCategoryId.Data4[6], auditCategoryId.Data4[7]);
+
+                    if(!strcasecmp(guid_string, guid_ObjectAccess)) {
+                        if(AuditEnumerateSubCategories(&auditCategoryId, FALSE, &pAuditSubCategoryGuids, &subCategoryCount) == FALSE) {
+                            goto error;
+                        }
+
+                        if(AuditQuerySystemPolicy(pAuditSubCategoryGuids, subCategoryCount, &pAuditPolicies) == FALSE) {
+                            goto error;
+                        }
+
+                        // Probe each subcategory in the category
+                        for(ULONG subcategoryIndex = 0; subcategoryIndex < subCategoryCount; subcategoryIndex++) {
+                            AUDIT_POLICY_INFORMATION currentPolicy = pAuditPolicies[subcategoryIndex];
+
+                            snprintf(guid_string, 40, "{%08lX-%04hX-%04hX-%02hhX%02hhX-%02hhX%02hhX%02hhX%02hhX%02hhX%02hhX}",
+                                                        currentPolicy.AuditSubCategoryGuid.Data1, currentPolicy.AuditSubCategoryGuid.Data2, currentPolicy.AuditSubCategoryGuid.Data3,
+                                                        currentPolicy.AuditSubCategoryGuid.Data4[0], currentPolicy.AuditSubCategoryGuid.Data4[1], currentPolicy.AuditSubCategoryGuid.Data4[2], currentPolicy.AuditSubCategoryGuid.Data4[3],
+                                                        currentPolicy.AuditSubCategoryGuid.Data4[4], currentPolicy.AuditSubCategoryGuid.Data4[5], currentPolicy.AuditSubCategoryGuid.Data4[6], currentPolicy.AuditSubCategoryGuid.Data4[7]);
+
+                            if(currentPolicy.AuditingInformation & POLICY_AUDIT_EVENT_SUCCESS) {
+                                if(!strcasecmp(guid_string, guid_FileSystem)) {
+                                    file_system_success = 1;
+                                }
+                                if(!strcasecmp(guid_string, guid_Handle)) {
+                                    handle_manipulation_success = 1;
+                                }
+                                if(file_system_success && handle_manipulation_success) {
+                                    LsaFreeMemory(AuditEvents);
+                                    LsaClose(PolicyHandle);
+
+                                    policies_checked++;
```

Comment:
This is a counter that is increasing every time the state checker thread runs, just to be used as a flag to know when it started to work.

There should be a cleaner way of doing this. A simple flag that turns from 0 to 1 would be enough.

On the other hand, we could try to run the state checker thread before starting to monitor the Eventchannel for the events. That way we don't have to keep saving the monitoring state in a variable.

Reply:
Done